lib/views/addToCollection.js

```diff
@@ -0,0 +1,119 @@
+const loadTemplate = require('../loadTemplate')
+const pug = require('pug')
+const config = require('../config')
+const sparql = require('../sparql/sparql')
+const getOwnedBy = require('../query/ownedBy')
+
+module.exports = function (req, res) {
+  req.setTimeout(0) // no timeout
+
+  function addToCollectionsForm (req, res) {
+    var collectionQuery = 'PREFIX dcterms: <http://purl.org/dc/terms/> PREFIX sbol2: <http://sbols.org/v2#> SELECT ?object ?name WHERE { ?object a sbol2:Collection . OPTIONAL { ?object dcterms:title ?name . }}'
+
+    function sortByNames (a, b) {
+      if (a.name < b.name) {
+        return -1
+      } else {
+        return 1
+      }
+    }
+
+    return sparql.queryJson(collectionQuery, req.user.graphUri).then((collections) => {
+      collections.forEach((result) => {
```

Comment:
A `.map()` might make more sense here than `forEach`.

Reply:
Ok, I switched, and it works.  Not sure I understand the difference here.  I think we are using this same structure in other places.